tdms/include/arrays/xyz_vector.h

```diff
@@ -0,0 +1,64 @@
+/**
+ * @file xyz_vector.h
+ * @author William Graham
+ * @brief Declares the XYZVector structure, which stores three vector members x,
+ * y, z.
+ */
+#pragma once
+
+#include <vector>
+
+#include "globals.h"
+
+/**
+ * @brief Structure containing 3 std::vector<double> members: x, y, z.
+ * @details Intended for use in instances where a quantity varies across the
+ * three axial directions, but the variation in each direction is independent of
+ * the other axial directions.
+ *
+ * Members can be referenced through the [] operator and passing the
+ * corresponding single-letter character.
+ */
+struct XYZVector {
+  std::vector<double> x = {};
+  std::vector<double> y = {};
+  std::vector<double> z = {};
+
+  std::vector<double> &operator[](const char &direction) {
+    switch (direction) {
+      case 'x':
+        return x;
+        break;
+      case 'y':
+        return y;
+        break;
+      case 'z':
+        return z;
+        break;
+      default:
+        throw std::runtime_error("Component not recognised");
+        break;
```

Comment:
```suggestion
      case 'x':
        return x;
      case 'y':
        return y;
      case 'z':
        return z;
      default:
        throw std::runtime_error("Component not recognised");
        break;
```
On older compilers, I claim we'd even get `Wunreachable-code` warnings.